e2etest/declarativeTestFiles.go

```diff
@@ -84,19 +84,20 @@ func (h *contentHeaders) String() string {
 // This is exposed to the declarativeResourceManagers, to create/check the objects.
 // All field are pointers or interfaces to make them nil-able. Nil means "unspecified".
 type objectProperties struct {
-	isFolder           bool // if false, the object is a file
-	size               *int64
-	contentHeaders     *contentHeaders
-	nameValueMetadata  map[string]string
-	blobTags           common.BlobTags
-	blobType           common.BlobType
-	creationTime       *time.Time
-	lastWriteTime      *time.Time
-	smbAttributes      *uint32
-	smbPermissionsSddl *string
-	adlsPermissionsACL *string // TODO: Test owner and group; needs a good target though.
-	cpkInfo            *common.CpkInfo
-	cpkScopeInfo       *common.CpkScopeInfo
+	isFolder              bool // if false, the object is a file
+	size                  *int64
+	contentHeaders        *contentHeaders
+	nameValueMetadata     map[string]string
+	blobTags              common.BlobTags
+	blobType              common.BlobType
+	creationTime          *time.Time
+	lastWriteTime         *time.Time
+	smbAttributes         *uint32
+	smbPermissionsSddl    *string
+	adlsPermissionsACL    *string // TODO: Test owner and group; needs a good target though.
+	cpkInfo               *common.CpkInfo
+	cpkScopeInfo          *common.CpkScopeInfo
+	s2sPreserveAccessTier string
```

Comment:
I can't see anywhere this is used for the creation of files or testing the files.

Reply:
It's being used in e2etest/zt_preserve_access_tier_test.go at ln 37 and ln 51.

Reply:
What I meant was, I don't see this actually _doing_ anything in the creation nor validation of files (in scenario_helpers.go:349 (scenarioHelper.generateBlobsFromList) or declarativeScenario.go:327 (scenario.validateProperties)). As such, I don't believe these tests are actually testing anything other than a normal copy with a classic account.

Additionally, `accessTier` may be a better name.